### PR TITLE
cassandra: ignore timeout in background jobs

### PR DESCRIPTION
**Pull request by iksaif**

### CODE REVIEW

File: biggraphite/drivers/cassandra.py [hunk 1]
```diff
@@ -2042,9 +2061,17 @@ def run(rows):
...
         token = start_token
         while token < stop_token:
-            rows = self._execute_metadata(select, (token,), DEFAULT_TIMEOUT_QUERY_UTIL)
-            if len(rows.current_rows) == 0:
-                break
+            try:
+                rows = self._execute_metadata(select, (token,), DEFAULT_TIMEOUT_QUERY_UTIL)
+
+                # Empty results means that we've reached the end.
+                if len(rows.current_rows) == 0:
```
Comment (wdauchy) [inline biggraphite/drivers/cassandra.py]: if I am not wrong you can simply write
if not rows.current_rows:

File: biggraphite/drivers/cassandra.py [hunk 2]
```diff
@@ -2001,11 +2010,22 @@ def clean(self, max_age=None, start_key=None, end_key=None, shard=1, nshards=0,
     def _prepare_background_request(self, query_str):
         select = self.__session_metadata.prepare(query_str)
         select.consistency_level = self._meta_background_consistency
-        select.retry_policy = cassandra.policies.DowngradingConsistencyRetryPolicy
+        select.retry_policy = cassandra.policies.DowngradingConsistencyRetryPolicy()
         select.request_timeout = DEFAULT_TIMEOUT_QUERY_UTIL
 
         return select
 
+    def _prepare_background_request_on_index(self, query_str):
```
Comment (Thib17) [inline biggraphite/drivers/cassandra.py]: Should it be named ...on_index, or ...always_retry ?
I really don't see where is it specific to a query on index.

File: biggraphite/drivers/cassandra.py [hunk 3]
```diff
@@ -2001,11 +2010,22 @@ def clean(self, max_age=None, start_key=None, end_key=None, shard=1, nshards=0,
...
+        select = self.__session_metadata.prepare(query_str)
+        select.retry_policy = bg_cassandra_policies.AlwaysRetryPolicy()
+        print(select)
```
Comment (Thib17) [inline biggraphite/drivers/cassandra.py]: remove print()

File: biggraphite/drivers/cassandra.py [hunk 4]
```diff
@@ -2001,11 +2010,22 @@ def clean(self, max_age=None, start_key=None, end_key=None, shard=1, nshards=0,
     def _prepare_background_request(self, query_str):
         select = self.__session_metadata.prepare(query_str)
         select.consistency_level = self._meta_background_consistency
-        select.retry_policy = cassandra.policies.DowngradingConsistencyRetryPolicy
+        select.retry_policy = cassandra.policies.DowngradingConsistencyRetryPolicy()
```
Comment (Thib17) [inline biggraphite/drivers/cassandra.py]: What is the impact ?

File: biggraphite/drivers/cassandra.py [hunk 5]
```diff
@@ -1807,9 +1808,17 @@ def map(self, callback, start_key=None, end_key=None, shard=1, nshards=0):
...
         token = start_token
         while token < stop_token:
-            rows = self._execute_metadata(select, (token,), DEFAULT_TIMEOUT_QUERY_UTIL)
-            if len(rows.current_rows) == 0:
-                break
+            try:
+                rows = self._execute_metadata(select, (token,), DEFAULT_TIMEOUT_QUERY_UTIL)
+
+                # Empty results means that we've reached the end.
+                if len(rows.current_rows) == 0:
```
Comment (wdauchy) [inline biggraphite/drivers/cassandra.py]: same possible comment below

File: biggraphite/drivers/cassandra.py (hunk 4 above)
Comment (iksaif) [reply biggraphite/drivers/cassandra.py]: The previous thing was a bug

File: biggraphite/drivers/cassandra.py (hunk 2 above)
Comment (iksaif) [reply biggraphite/drivers/cassandra.py]: It prepares a background request that will use an index.

Index queries have specific needs.

File: biggraphite/drivers/cassandra.py (hunk 3 above)
Comment (iksaif) [reply biggraphite/drivers/cassandra.py]: done

File: biggraphite/drivers/cassandra.py (hunk 1 above)
Comment (iksaif) [reply biggraphite/drivers/cassandra.py]: I'd like to not change this code (I just moved it). I think I remember there was a reason for the len() but I'm not sure which one.

File: biggraphite/drivers/cassandra.py [hunk 6]
```diff
@@ -1994,18 +2017,28 @@ def clean(self, max_age=None, start_key=None, end_key=None, shard=1, nshards=0,
         """
         super(_CassandraAccessor, self).clean(max_age, callback_on_progress)
 
-        self._clean_empty_dir(start_key, end_key, shard, nshards, callback_on_progress)
+        #self._clean_empty_dir(start_key, end_key, shard, nshards, callback_on_progress)
```
Comment (InformatiQ) [inline biggraphite/drivers/cassandra.py]: why comment that line? if it is not needed delete it

File: biggraphite/drivers/cassandra.py (hunk 1 above)
Comment (wdauchy) [reply biggraphite/drivers/cassandra.py]: ok

File: biggraphite/drivers/cassandra.py (hunk 6 above)
Comment (iksaif) [reply biggraphite/drivers/cassandra.py]: typo

**Pull request merged**